### ISSUE TITLE
jacoco + junit5 + tesable-mock org.junit.platform.commons.JUnitException: TestEngine with ID 'junit-vintage' failed to discover tests 

### ISSUE THREAD

**Issue by buyaonitongle** [alibaba/testable-mock#56]
# Created at 2021-01-29T10:23:50.413
```
[org.junit.platform.commons.JUnitException: TestEngine with ID 'junit-vintage' failed to discover tests
at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:111)
	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:85)
	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:92)
	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:67)
	at org.apache.maven.surefire.junitplatform.TestPlanScannerFilter.accept(TestPlanScannerFilter.java:56)
	at org.apache.maven.surefire.util.DefaultScanResult.applyFilter(DefaultScanResult.java:102)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.scanClasspath(JUnitPlatformProvider.java:146)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:127)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
Caused by: java.lang.VerifyError: Bad type on operand stack
Exception Details:
  Location:
    xxx/SmnObserverTest.getAppGlobalDispatcher()Lscala/concurrent/ExecutionContextExecutor; @18: invokespecial
  Reason:
    Type '[Z' (current frame, stack[2]) is not assignable to 'xxx/SmnObserverTest'
  Current Frame:
    bci: @18
    flags: { }
    locals: { '[Z' }
    stack: { uninitialized 13, uninitialized 13, '[Z' }
  Bytecode:
    0x0000000: 03bd 0004 0303 b800 38b8 002b 4bbb 0013
    0x0000010: 592a b700 3b2b 0404 54b0              
at java.lang.Class.getDeclaredMethods0(Native Method)
	at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)
	at java.lang.Class.privateGetPublicMethods(Class.java:2902)
	at java.lang.Class.getMethods(Class.java:1615)
	at org.junit.platform.commons.util.ReflectionUtils.getDefaultMethods(ReflectionUtils.java:1481)
	at org.junit.platform.commons.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:1454)
	at org.junit.platform.commons.util.ReflectionUtils.findAllMethodsInHierarchy(ReflectionUtils.java:1396)
	at org.junit.platform.commons.util.ReflectionUtils.findMethods(ReflectionUtils.java:1380)
	at org.junit.platform.commons.util.ReflectionUtils.findMethods(ReflectionUtils.java:1366)
	at org.junit.vintage.engine.descriptor.TestSourceProvider.lambda$findMethod$1(TestSourceProvider.java:64)
	at java.util.HashMap.computeIfAbsent(HashMap.java:1126)
	at java.util.Collections$SynchronizedMap.computeIfAbsent(Collections.java:2672)
	at org.junit.vintage.engine.descriptor.TestSourceProvider.findMethod(TestSourceProvider.java:64)
	at org.junit.vintage.engine.descriptor.TestSourceProvider.findTestSource(TestSourceProvider.java:45)
	at org.junit.vintage.engine.discovery.RunnerTestDescriptorPostProcessor.addChildrenRecursively(RunnerTestDescriptorPostProcessor.java:63)
	at org.junit.vintage.engine.discovery.RunnerTestDescriptorPostProcessor.applyFiltersAndCreateDescendants(RunnerTestDescriptorPostProcessor.java:45)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at org.junit.vintage.engine.discovery.VintageDiscoverer.discover(VintageDiscoverer.java:50)
	at org.junit.vintage.engine.VintageTestEngine.discover(VintageTestEngine.java:63)
	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:103)
	... 11 more]([url](`url`))

```
I just use @MockMethod like this:

  ```
  @MockMethod(targetClass = SystemHolder.class)
    private ExecutionContextExecutor getAppGlobalDispatcher() {
        return new ExecutionContextExecutor() {
            @Override
            public void execute(Runnable runnable) {
                runnable.run();
            }

            @Override
            public void reportFailure(Throwable cause) {

            }

            @Override
            public ExecutionContext prepare() {
                return ExecutionContextExecutor.super.prepare();
            }
        };
    }
```

**Comment by linfan** [comment]
Mock methods with anonymous class currently have to be defined as `static`. This is a compatibility trade-off of design in `v0.4.x` series, which will be finally fix in upcoming `v0.5` version.

Check [additional note](https://alibaba.github.io/testable-mock/#/en-us/doc/use-mock?id=additional-note) for more detail.

If that didn't fix your issue, I'll go deeper to check.

**Comment by buyaonitongle** [comment]
`
    @MockMethod(targetClass = SystemHolder.class)
    private static ExecutionContextExecutor getAppGlobalDispatcher() {
        return new ExecutionContextExecutor() {
            @Override
            public void execute(Runnable runnable) {
                runnable.run();
            }

            @Override
            public void reportFailure(Throwable cause) {

            }

            @Override
            public ExecutionContext prepare() {
                return ExecutionContextExecutor.super.prepare();
            }
        };
    }
`
fixed. ok
